### PR TITLE
Add Pulse Trigger to push MetaTask when platform build's MD5 is changed

### PR DESCRIPTION
**Pull request by askeing**
1. `jobs/pulse.py` will be loaded by `ejenti`.

2. `pulse_modules/*MetaTasks.py` will handle the MetaTasks.
Publisher will send the MetaTasks object to Consumer by Pulse.
And Consumer (jobs.pulse) will loads it and get the information from this object.

3. `pulse_modules/hasalPulsePublisher.py` handle the action of pushing MetaTasks object.

4. `pulse_modules/tasksTrigger.py` will check the platform's build MD5.
If there is new platform's build, then push MetaTasks.
It will also declare the queue, or the message will be ignored if queue doesn't exist.

5. `pulse_trigger.py` is trigger entry point.
```bash
pulse_trigger.py [--config=<str>] [--cmd-config=<str>]
 --config=<str>      Specify the trigger_config.json file path. [default: trigger_config.json]
 --cmd-config=<str>  Specify the cmd_config.json file path. [default: cmd_config.json]
```

### CODE REVIEW

**Comment by askeing** [comment]
@ShakoHo @MikeLien @ypwalter r?

File: ejenti/pulse_modules/tasksTrigger.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+                                                                 topic=topic)
+            if not queue_exists:
+                logging.error('There is not Queue for Topic [{topic}]. Message might be ignored.'.format(topic=topic))
+
+            # Push MetaTask to Pulse
+            publisher = HasalPulsePublisher(username=username,
+                                            password=password,
+                                            command_config=command_config)
+
+            uid_prefix = '{time}.{job}'.format(time=datetime.now(), job=job_name)
```
Comment (ShakoHo) [inline ejenti/pulse_modules/tasksTrigger.py]: how about using a better format without empty space?

**Comment by askeing** [comment]
@ShakoHo r? again

File: ejenti/pulse_modules/tasksTrigger.py [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+                logging.info('Job "{}" platform "{}": Latest Hash [{}], Origin Hash: [{}]'.format(job_name,
+                                                                                                  platform,
+                                                                                                  new_hash,
+                                                                                                  origin_hash))
+                with open(job_md5_file, 'w') as f:
+                    f.write(new_hash)
+                return True
+            pass
+        else:
+            # 1st check
```
Comment (ypwalter) [inline ejenti/pulse_modules/tasksTrigger.py]: 1st check?

File: ejenti/pulse_modules/tasksTrigger.py [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+            else:
+                # changed
+                logging.info('Job "{}" platform "{}": Latest Hash [{}], Origin Hash: [{}]'.format(job_name,
+                                                                                                  platform,
+                                                                                                  new_hash,
+                                                                                                  origin_hash))
+                with open(job_md5_file, 'w') as f:
+                    f.write(new_hash)
+                return True
+            pass
```
Comment (ypwalter) [inline ejenti/pulse_modules/tasksTrigger.py]: what is this `pass` for?

File: ejenti/pulse_modules/tasksTrigger.py [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+            'key': 'mac',
+            'ext': 'dmg'
+        },
+        'win32': {
+            'key': 'win32',
+            'ext': 'zip'
+        },
+        'win64': {
+            'key': 'win64',
+            'ext': 'zip',
```
Comment (ypwalter) [inline ejenti/pulse_modules/tasksTrigger.py]: remove the "," in line 55

**Pull request merged**